firmware/user/display/oled.c

```diff
@@ -202,6 +202,213 @@ void drawPixel(int16_t x, int16_t y, color c)
     }
 }
 
+
+/**
+ * Set/clear/invert a single pixel.
+ *
+ * This intentionally does not have ICACHE_FLASH_ATTR because it may be called often
+ *
+ * @param x Column of display, 0 is at the left
+ * @param y Row of the display, 0 is at the top
+ */
+
+void drawPixelFastWhite( int x, int y )
+{
+    uint8_t* addy = &currentFb[(y + x * OLED_HEIGHT) / 8];
+    uint8_t mask = 1 << (y & 7);
+    *addy |= mask;
+}
+
+/**
+ * @brief Optimized method to quickly draw a white line.
+ *
+ * @param x1, x0 Column of display, 0 is at the left
+ * @param y1, y0 Row of the display, 0 is at the top
+ *
+ */
+
+
+#define LABS( x ) (((x)<0)?-(x):(x))
+
+//This is fast, but broken :(
+void ICACHE_FLASH_ATTR speedyWhiteLine( int16_t x0, int16_t y0, int16_t x1, int16_t y1 )
+{
+//Tune this as a function of the size of your viewing window, line accuracy, and worst-case scenario incoming lines.
+#define BRESEN_W OLED_WIDTH
+#define BRESEN_H OLED_HEIGHT
+#define FIXEDPOINT 65536
+#define FIXEDPOINTD2 (FIXEDPOINT/2)
+    int dx = (x1-x0);
+    int dy = (y1-y0);
+    int sdx = (dx>0)?1:-1;
+    int sdy = (dy>0)?1:-1;
+    int x = x0;
+    int y = y0;
+
+    int yerrdiv = ( dx * sdx );  //dy, but always positive.
+    int xerrdiv = ( dy * sdy );  //dx, but always positive.
+    int yerrnumerator;
```

Comment:
Compiler is warning that `xerrnumerator` and `yerrnumerator` may not be initialized before use. Worth setting to zero here?

Reply:
That is in fact not possible.  In addition, the performance is measurably slower when these are set on initialization. 23.461ms vs 23.532ms. 

Also, now that I've started here WHY IS DIVIDING A NUMBER BY 65536 SLOWER THAN SHIFTING IT BY 16?!?!?!

Reply:
I mean I'll update it but :sigh:

Reply:
Compilers are dumber than humans. Warnings like that have caught edge cases for me enough times that I just go with them now. Division v. bitshifting seems inexcusable tho.